### ISSUE TITLE
ability to generate mocks from _test.go files

### ISSUE THREAD

**Issue by ormanli** [vektra/mockery#700]
Description
-------------

I want to create a single mock for all of the interfaces I am using. I am creating an interface that embeds all of the interfaces in a test file. But when I want to generate a mock for that interface mockery fails. But if I move that interface to a non-test file , it generates without any problem.

Mockery Version
--------

v2.33.1

Golang Version
--------------

Go 1.20.6

Installation Method
-------------------

- [ ] Binary Distribution
- [ ] Docker
- [ ] brew
- [x] go install
- [ ] Other: [specify]

Steps to Reproduce
------------------

1. [First Step]
Create a file.
```
import (
	"fmt"
	"io"
	"testing"
)

//go:generate mockery --with-expecter=true --inpackage=true --testonly=true --name=testDependencies

type testDependencies interface {
	io.Reader
	fmt.Stringer
}

func TestName(t *testing.T) {
}
```
3. [Second Step]
run `go generate`

### Expected Behavior

Generate a mock.

### Actual Behavior

```
31 Aug 23 19:07 +03 INF couldn't read any config file version=v2.33.1
31 Aug 23 19:07 +03 INF Starting mockery dry-run=false version=v2.33.1
31 Aug 23 19:07 +03 INF Using config:  dry-run=false version=v2.33.1
31 Aug 23 19:07 +03 WRN DEPRECATION: use of the packages config will be the only way to generate mocks in v3. Please migrate your config to use the packages feature. dry-run=false migration=https://vektra.github.io/mockery/v2.33/migrating_to_packages/ url=https://vektra.github.io/mockery/v2.33/features/#packages-configuration version=v2.33.1
31 Aug 23 19:07 +03 INF Walking dry-run=false version=v2.33.1
31 Aug 23 19:07 +03 ERR Unable to find 'testDependencies' in any go files under this path dry-run=false version=v2.33.1
unable to find interface
a_test.go:9: running "mockery": exit status 1
```

**Comment by LandonTClipp** [comment]
The `packages.Config` struct has a `Tests` bool that if we set to true, _might_ include parsing of the test files.

https://pkg.go.dev/golang.org/x/tools/go/packages#Config
https://github.com/vektra/mockery/blob/v2.33.1/pkg/parse.go

If you don't mind, could you try modifying that and build mockery from source to see if it does what you want?

**Comment by ormanli** [comment]
Hi,
I modified the source code by setting `conf.Tests` to true in `NewParser` function. Now it panics because there are multiple packages from line https://github.com/vektra/mockery/blob/v2.33.1/pkg/parse.go#L152. But in the message it shows the same package twice.

```
panic: file /Users/ormanli/repo/mockery_test/config/config.go resolves to multiple packages: config, config
```

**Comment by LandonTClipp** [comment]
Could you try this experiment by using the `packages` config feature? I think the particular code path you're hitting (which is deprecated BTW) would have to be modified to select the package you need. The `Tests` documentation says this:

```
	// If Tests is set, the loader includes not just the packages
	// matching a particular pattern but also any related test packages,
	// including test-only variants of the package and the test executable.
	//
	// For example, when using the go command, loading "fmt" with Tests=true
	// returns four packages, with IDs "fmt" (the standard package),
	// "fmt [fmt.test]" (the package as compiled for the test),
	// "fmt_test" (the test functions from source files in package fmt_test),
	// and "fmt.test" (the test binary).
	//
	// In build systems with explicit names for tests,
	// setting Tests may have no effect.
```

so we'd probably need to select your `config [config.test]` package and discover the interfaces defined in there. I'm not sure there's a clean way to do that with the legacy config semantics you're using. If we do have to make some special logic in the config to select which test package you want to parse, it'd be much easier to specify in the `packages` config.

**Comment by ormanli** [comment]
I tried `[fmt.test]` and `fmt_test` with new packages config yaml. Both of them failed with `no required module provides package github.com/ormanli/mockery_test/config_test` and `no required module provides package github.com/ormanli/mockery_test/[config.test]` .

The code base I am working on is defining and using interfaces like Java, next to implementation; instead of next to who uses it like in Go. I decided to merge all of the used interfaces in my test to generate a single mock file.

I will generate separate files for each of the interfaces. The new packages config is working for that case. We can close this issue.
